### PR TITLE
feat: remove caseworkers family name on `selfserve`

### PR DESCRIPTION
**Pull request by gabrielg2020**
## Description
remove caseworkers family name in messages on `selfserve`

## Before submitting (or marking as "ready for review")

- [x] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [x] Have you performed a self-review of the code
- [ ] Have you have added tests that prove the fix or feature is effective and working
- [ ] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

**Review by fibble — APPROVED** [review]

**Review by fibble — COMMENTED** [review]
In refinement it was mentioned that the caseworkers name in full may end up on a snapshot available to self-serve users. Is that being dealt with separately?

**Comment by gabrielg2020** [comment]
@fibble 
> In refinement it was mentioned that the caseworkers name in full may end up on a snapshot available to self-serve users. Is that being dealt with separately?

I am not sure. I will message Kab to find out!

File: Common/src/Common/Service/Table/Formatter/AbstractConversationMessage.php [hunk 1]
```diff
@@ -33,6 +33,13 @@ public function format($row, $column = null): string
         // If createdBy (User) has a Team, they are an internal user.
         $internalCaseworkerTeam = (empty($row['createdBy']['team'])) ? '' : '<p class="govuk-caption-m">' . $senderName . '<br/>Caseworker Team</p>';
 
+        // If running on 'selfserve' remove caseworker's family name.
+        if (str_contains(__FILE__, 'selfserve') && $internalCaseworkerTeam) {
+            $caseworkerFamilyName = explode(' ', $senderName)[1];
+            $senderName = str_replace($caseworkerFamilyName, '', $senderName);
+            $internalCaseworkerTeam = str_replace($caseworkerFamilyName, '', $internalCaseworkerTeam);
+        }
+
```
Comment (JoshuaLicense) [inline Common/src/Common/Service/Table/Formatter/AbstractConversationMessage.php:42]: The `__FILE__` magic constant relies on a file path that contains `selfserve` which is not going always the case. For example, the file path I have for this file is: `/var/www/html/vendor/olcs/olcs-common/Common/src/Common/Service/Table/Formatter/AbstractConversationMessage.php`.

Instead, I recommend taking advantage of the inheritance that's going on in this area. The frontend applications will each have a concrete class that extends this _abstract_ class. Each one of those concrete classes could have its own `getSenderName` that formats appropriately. 

A few unit tests for this logic would also be ideal!

Give me a shout if you need a hand.

**Review by JoshuaLicense — CHANGES_REQUESTED** [review]

**Comment by gabrielg2020** [comment]
Removing family name on snapshot is a PR on `backend`: https://github.com/dvsa/olcs-backend/pull/205

**Review by JoshuaLicense — APPROVED** [review]

**Review by fibble — APPROVED** [review]

**Pull request merged**